### PR TITLE
twitch-chat-downloader: update client_id

### PR DESCRIPTION
**Pull request by Assistant**
## Description of changes

From [this PR](https://github.com/TheDrHax/Twitch-Chat-Downloader/pull/16).
This allows the program to function again after changes to the Twitch backend.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/twitch-chat-downloader/default.nix [hunk 1]
```diff
@@ -20,6 +21,14 @@ buildPythonApplication rec {
     hash = "sha256-mV60ygrtQa9ZkJ2CImhAV59ckCJ7vJSA9cWkYE2xo1M=";
   };
 
+  patches = [
+    # Updates client_id
+    (fetchpatch {
+      url = "https://github.com/TheDrHax/Twitch-Chat-Downloader/commit/23de61a0221d3796b706774283111734dd2338e6.patch";
+      hash = "sha256-qwFRyV1MeC4Isc2gVEYStr6ytP7EzN5WuZOwQ8KOqTc=";
+    })
```
Comment (emilazy) [inline pkgs/applications/misc/twitch-chat-downloader/default.nix:29]: We generally like to avoid `fetchpatch`ing unmerged PRs since GitHub can garbage collect the commits. Usually we would vendor the `.patch` file in the tree instead, but since this is so trivial you could just put `substituteInPlace tcd/example.settings.json --replace-fail old_client_id new_client_id` in a `postPatch` phase instead.

**Review by emilazy — CHANGES_REQUESTED** [review]
Builds on `x86_64-darwin`, unsurprisingly.

File: pkgs/applications/misc/twitch-chat-downloader/default.nix (hunk 1 above)
Comment (Assistant) [reply pkgs/applications/misc/twitch-chat-downloader/default.nix:29]: That is surprising, I've mostly come across `fetchpatch`es of unmerged PRs, and actually had my vendored `.patch` replaced with one in another package. 
`substituteInPlace` makes a lot more sense here.

Comment (emilazy) [reply pkgs/applications/misc/twitch-chat-downloader/default.nix]: Yeah, unfortunately the pitfalls of unmerged GitHub PRs aren‘t too widely known. I had to be convinced myself recently :)

File: pkgs/applications/misc/twitch-chat-downloader/default.nix [hunk 2]
```diff
@@ -20,6 +20,11 @@ buildPythonApplication rec {
     hash = "sha256-mV60ygrtQa9ZkJ2CImhAV59ckCJ7vJSA9cWkYE2xo1M=";
   };
 
+  postPatch = ''
+    substituteInPlace tcd/example.settings.json --replace-fail \
+    jzkbprff40iqj646a697cyrvl0zt2m6 kd1unb4b3q4t58fwlpcbzcbnm76a8fp
+  '';
```
Comment (emilazy) [inline pkgs/applications/misc/twitch-chat-downloader/default.nix:26]: Could you link https://github.com/TheDrHax/Twitch-Chat-Downloader/pull/16 in a comment for reference as to why this change is being made?

```suggestion
  postPatch = ''
    # Update client ID for Twitch changes
    # See: <https://github.com/TheDrHax/Twitch-Chat-Downloader/pull/16>
    substituteInPlace tcd/example.settings.json \
      --replace-fail jzkbprff40iqj646a697cyrvl0zt2m6 kd1unb4b3q4t58fwlpcbzcbnm76a8fp
  '';
```

**Review by emilazy — APPROVED** [review]
Looks good, thanks!
